src/CruxComponent.tsx

```diff
@@ -190,35 +194,47 @@ export class CruxComponentCreator {
                                     {constants.editModal && <th></th>}
                                 </tr>
                                 </thead>
                                 <tbody>
-                                {_.map(filteredRows,
-                                    (model: any, index: number) => {
-                                        const filtered = constants.fields.filter((field: any) => field.display === true)
-                                        return <tr key={index}>
-                                            {_.map(filtered, (field: any, i: number) => {
-                                                    return <td key={i}
-                                                               style={(field.cellCss) ? field.cellCss : { margin: "0px" }}>
-                                                        <div style={{ marginTop: 8 }}>
-                                                            <ListNestedComponent
-                                                                field={field} model={model}
-                                                                additionalModels={this.props.additionalModels}
-                                                                modelChanged={this.inlineEdit} />
-                                                        </div>
-                                                    </td>
-                                                }
-                                            )}
-                                            {constants.editModal &&
-                                            <td key={2}><span style={{ marginLeft: "20px", marginTop: 8, color: "grey" }}
-                                                              className="glyphicon glyphicon-pencil fas fa-pencil-alt"
-                                                              aria-hidden="true" onClick={this.showEditModal(model)} />
-                                            </td>}
-                                        </tr>
+                                <tr key='searchRow'>
+                                    {constants.fields.filter((field: any) => field.display === true).map((field: any, i: number) => {
+                                        return <td key={i} style={(field.cellCss) ? field.cellCss : { margin: "0px" }}>
+                                            <div>
+                                                {field.search && <FormGroup>
+                                                    <FormControl type="text"
+                                                        value={(this.state.filterModel || {})[field.search.key]}
+                                                        onChange={(e: any) => this.handleFieldSearch(field.search.key, e.target.value)}
+                                                        onBlur={(e: any) => this.props.filter(constants.modelName, this.state.filterModel, this.filterSuccess)}
```

Comment:
trigger server-side filtering on focus out of search field

Reply:
Client should be able to control through config whether server side filtering or client side filtering is needed